ui/app/components/tools/wrap.ts

```diff
@@ -51,37 +57,37 @@ export default class ToolsWrap extends Component {
   reset(clearData = true) {
     this.token = '';
     this.errorMessage = '';
-    this.wrapTTL = null;
+    this.wrapTTL = '';
     this.hasLintingErrors = false;
     if (clearData) this.wrapData = null;
   }
 
   @action
-  updateTtl(evt) {
+  updateTtl(evt: TtlEvent) {
     if (!evt) return;
     this.wrapTTL = evt.enabled ? `${evt.seconds}s` : '30m';
   }
 
   @action
-  codemirrorUpdated(val, codemirror) {
+  codemirrorUpdated(val: string, codemirror: Editor) {
     codemirror.performLint();
     this.hasLintingErrors = codemirror?.state.lint.marked?.length > 0;
     if (!this.hasLintingErrors) this.wrapData = JSON.parse(val);
   }
 
   @action
-  async handleSubmit(evt) {
+  async handleSubmit(evt: HTMLElementEvent<HTMLFormElement>) {
     evt.preventDefault();
 
-    const data = this.wrapData;
-    const wrapTTL = this.wrapTTL || null;
+    const data = this.wrapData || {};
+    const wrap = this.wrapTTL || '';
 
     try {
-      const response = await this.store.adapterFor('tools').toolAction('wrap', data, { wrapTTL });
-      this.token = response.wrap_info.token;
+      const { wrap_info } = await this.api.sys.wrap(data, this.api.buildHeaders({ wrap }));
+      this.token = wrap_info?.token || '';
       this.flashMessages.success('Wrap was successful.');
     } catch (error) {
-      this.errorMessage = errorMessage(error);
+      this.errorMessage = await apiErrorMessage(error);
```

Comment:
Do you need to manipulate the api error at all. Ex `[here](https://github.com/hashicorp/vault/blob/main/ui/app/components/secret-engine/configure-wif.ts#L184)`.